log/brimapi/client.go

```diff
@@ -0,0 +1,103 @@
+package brimapi
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/allegro/akubra/service"
+	"github.com/hashicorp/consul/api"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	uploadSynctasksURI         = "/v1/processes/uploadsynctasks"
+	serviceDiscoverySchemeName = "service"
+	httpRequestTimeout         = time.Second * 3
+	httpConsulRequestTimeout   = time.Second * 2
+)
+
+var (
+	discoveryServices *service.Services
+	httpClient        *http.Client
+)
+
+func init() {
+	httpClient = &http.Client{
+		Timeout: httpRequestTimeout,
+	}
+	consulConfig := api.DefaultConfig()
+	api.DefaultConfig().Transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   httpConsulRequestTimeout,
+			DualStack: true,
+		}).DialContext,
+	}
+	consulClient, err := api.NewClient(consulConfig)
+	if err != nil {
+		fmt.Printf("unable to create Consul client: %s", err)
+	}
+
+	discoveryServices = service.New(consulClient, service.DefaultCacheInvalidationTimeout)
+}
+
+func doRequest(lh *LogHook, httpClient *http.Client, payload string) (endpoint string, err error) {
```

Comment:
Why these methods (this and the one below)  accept a `payload`? What if I wanted to send a stream of bytes?